py/server/deephaven/table_listener.py

```diff
@@ -442,3 +445,173 @@ def listen(t: Table, listener: Union[Callable, TableListener], description: str
+
+        Returns:
+            a bool
+        """
+        return self.j_listener_recorder.recordedVariablesAreValid()
+
+    def table_update(self) -> Optional[TableUpdate]:
+        """Gets the table update from the listener recorder. If there is no update in the current update graph cycle,
+        returns None.
+
+        Returns:
+            a TableUpdate or None
+        """
+        j_table_update = self.j_listener_recorder.getUpdate()
+        return TableUpdate(self.table, j_table_update) if j_table_update else None
+
+
+class MergedListener(ABC):
+    """A MergedListener has a collection of ListenerRecorder. Each one must complete before the merged listener
+    executes only once for the current update graph cycle. """
+
+    listener_recorders: Sequence[ListenerRecorder]
+
+    @abstractmethod
+    def process(self) -> None:
+        """The required method on a user defined subclass of MergedListener that processes the table updates from the
+        list of ListenerRecorders automatically provided after being adapted by the MergedListenerHandle.
+        The implementation of this method in the subclass can reference the listener recorders by the
+        'listener_recorders' attribute. The listener recorders are in the same order as the source tables provided during
+        the instance initialization.
+
+        Example:
+            def process(self) -> None:
+                for i, listener in enumerate(self.listener_recorders):
+                    print(f"table {i}:", self.listener_recorders[i].table_update())
+        """
+        ...
+
+
+class MergedListenerHandle:
+    """A handle to manage a merged listener's lifecycle."""
+
+    def __init__(self, listener_recorders: Sequence[ListenerRecorder], listener: Union[Callable, MergedListener],
+                 description: str = None, dependencies: Union[Table, Sequence[Table]] = None):
+        """Creates a new MergedListenerHandle with the provided listener recorders and dependencies.
+
+        Table change events are processed by 'listener', which can be either
+        (1) a callable (e.g. function) or
+        (2) an instance of MergedTableListener type which provides a "process" method.
+        The callable or the process method must have zero parameters.
+
+        Note: Don't do table operations in the listener. Do them beforehand, and add the results as dependencies.
+
+        Args:
+            listener_recorders (Sequence[ListenerRecorder]): listener recorders that record the table updates for the
+                merged listener to process.
+            listener (Union[Callable, MergedListener]): listener to process table updates from the listener recorders
+            description (str, optional): description for the UpdatePerformanceTracker to append to the listener's entry
+            dependencies (Union[Table, Sequence[Table]]): tables that must be satisfied before the listener's execution.
```

Comment:
To be clear, these are *additional* dependencies. The recorders are always included unless you override methods from the superclass.